contracts/Everlong.sol

```diff
@@ -365,6 +408,66 @@ contract Everlong is IEverlong {
         return output;
     }
 
+    /// @dev Calculate the losses incurred from closing sufficient positions
+    ///      to receive at least `_targetOutput`.
+    /// @param _targetOutput Target asset output from closed positions.
+    /// @return losses Losses incurred from closing the immature positions.
+    function _calcImmaturePositionLosses(
+        uint256 _targetOutput
+    ) internal view returns (uint256 losses) {
+        // Initialize variables.
+        uint256 output;
+        uint256 proceeds;
+        uint256 estimatedProceeds;
+        IEverlong.Position memory position;
+        uint256 i;
+        uint256 count = _portfolio.positionCount();
+
+        // Retrieve the `weightedSpotPrice` from the most recent checkpoint.
+        uint256 spotPrice = IHyperdrive(hyperdrive)
+            .getCheckpointDown(block.timestamp)
+            .weightedSpotPrice;
+
+        // Iterate through the position queue (most to least mature).
+        // For each position, calculate the `closeLong` output at the current
+        // spot price and compare it to the `closeLong` output using the
+        // most recent checkpoint's `weightedSpotPrice`. The difference
+        // between the two is the loss.
+        while (i < count && output < _targetOutput) {
+            // Retrieve the position at the current index.
+            position = _portfolio.at(i);
+
+            // Calculate the `closeLong` output using the current spot price.
+            proceeds = IHyperdrive(hyperdrive).previewCloseLong(
+                asBase,
+                position,
+                ""
+            );
+
+            // Calculate the `closeLong` output using the latest checkpoint's
+            // `weightedSpotPrice` and add it to the received output.
+            estimatedProceeds = IHyperdrive(hyperdrive).previewCloseLong(
+                asBase,
+                position,
+                spotPrice,
+                ""
+            );
+            // TODO: Reevaluate whether `estimatedProceeds` or `proceeds`
+            //       should be used to increment output. Maybe use the min?
+            output += estimatedProceeds;
+
+            // If actual proceeds are less than estimated, add the difference
+            // to the cumulative losses.
+            if (proceeds < estimatedProceeds) {
+                losses += estimatedProceeds - proceeds;
+            }
```

Comment:
I was expecting to see us calculate closing the long using "mark-to-market accounting" (i.e. by using the weighted average spot price as the price) and then to see us calculate closing the long with a simulation. Then we could take the difference. As mentioned elsewhere, that's not really what we're doing here.

Reply:
I think we covered this offline but let me know if not